esFrontLine/app.py

```diff
@@ -67,11 +68,13 @@ def catch_all_get(path):
 def catch_all_post(path):
     return catch_all(path, 'post')
 
-
 def catch_all(path, type):
     try:
         data = flask.request.environ['body_copy']
-        filter(type, path, data)
+        resource = filter(type, path, data)
+
+        # Check HAWK authentication before processing request
+        auth.check(flask.request, resource)
```

Comment:
I would like the hawk authentication done before the `filter()`, if only to avoid processing the body when the user is not authenticated.  But I see the `resource` is used as part of the auth, which is nice.  I have no suggestion for you.